website/project/views/comment.py

```diff
@@ -198,41 +307,85 @@ def is_reply(target):
+
+    if page == 'total' and not root_id:  # "Total" on discussion page
+        comments = list_total_comments(node, auth, 'total')
+    elif page == 'total':  # Discussion widget on overview's page
+        comments = list_total_comments_widget(node, auth)
+        is_widget = True
+    elif not root_id:  # Overview/Files/Wiki page on discussion page
+        comments = list_total_comments(node, auth, page)
+    else:
+        target = resolve_target(node, page, guid)
+        comments = getattr(target, 'commented', [])
+
+    if is_list:
+        discussions = comment_discussion(comments, node, anonymous=anonymous, widget=is_widget)
 
+    n_unread = 0
     if node.is_contributor(auth.user):
         if auth.user.comments_viewed_timestamp is None:
             auth.user.comments_viewed_timestamp = {}
             auth.user.save()
-        n_unread = n_unread_comments(target, auth.user)
-    return {
-        'comments': serialized_comments,
+        n_unread = n_unread_comments(node, auth.user, page, root_id)
+
+    ret = {
+        'comments': [
+            serialize_comment(comment, auth, anonymous)
+            for comment in comments
+        ],
         'nUnread': n_unread
     }
+    if is_list:
+        ret.update(discussions)
+    return ret
+
+
+def list_total_comments_widget(node, auth):
+    comments_keys = Comment.find(Q('node', 'eq', node)).get_keys()
+    comments = []
+    for cid in comments_keys:
+        cmt = Comment.load(cid)
+        comments.append(cmt)
+    comments.sort(
+        key=lambda item: item.date_created,
+        reverse=False
+    )
+    return comments
 
 
-def n_unread_comments(node, user):
-    """Return the number of unread comments on a node for a user."""
-    default_timestamp = datetime(1970, 1, 1, 12, 0, 0)
-    view_timestamp = user.comments_viewed_timestamp.get(node._id, default_timestamp)
-    return Comment.find(Q('node', 'eq', node) &
-                        Q('user', 'ne', user) &
-                        Q('date_created', 'gt', view_timestamp) &
-                        Q('date_modified', 'gt', view_timestamp)).count()
+def list_total_comments(node, auth, page):
+    if page == 'total':
+        comments_keys = Comment.find(Q('node', 'eq', node)).get_keys()
+    else:
+        comments_keys = Comment.find(Q('node', 'eq', node) &
+                                Q('page', 'eq', page)).get_keys()
+    comments = []
+    for cid in comments_keys:
+        cmt = Comment.load(cid)
+        if not isinstance(cmt.target, Comment):
+            comments.append(cmt)
+    comments = sorted(
+        comments,
+        key=lambda item: item.date_created,
+        reverse=False,
+    )
+    return comments
 
 
 @must_be_logged_in
 @must_be_contributor_or_public
 def edit_comment(**kwargs):
-
     auth = kwargs['auth']
 
     comment = kwargs_to_comment(kwargs, owner=True)
 
-    content = request.json.get('content').strip()
+    content = request.get_json().get('content').strip()
```

Comment:
`.get('content')` can return `None` here, in which case we'll get an `AttributeError` when calling `.strip()`. This should be handled.